src/main/java/stringAddCalculate/StringSplit.java

```diff
@@ -0,0 +1,19 @@
+package stringAddCalculate;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringSplit {
+    private static final String DEFAULT_DELIMITER = "[,:]";
+    private static final String CUSTOM_DELIMITER = "//(.)\n(.*)";
+
+    public static String[] split(final String expression) {
+        Matcher m = Pattern.compile(CUSTOM_DELIMITER).matcher(expression);
```

Comment:
Pattern은 비싼 객체(객체를 만드는데 비용이 많이 듬)기 때문에, 아래와 같이 상수로 정의해서 사용하는 건 어떨까요?

```suggestion
    private static final Pattern SPLIT_KEY_PATTERN = Pattern.compile("//(.)\n(.*)");
```

```java
public class RomanNumber {

    private static final Pattern ROMAN = Pattern.compile("^(?=.)M*(C[MD]|D?C{0,3})(X[CL]|L?X{0,3})(I[XV]|V?I{0,3})$");

    static boolean isRomanNumeral(String s) {
        return ROMAN.matcher(s).matches();
    }
}
```

링크 : https://github.com/keesun/study/blob/master/effective-java/item6.md